### PR TITLE
Simulation option

### PR DESCRIPTION
**Pull request by KevinAVR**
Added simulatio option to all codes. There was a typos in state tomography and allxy for the strem processing that I fixed. Also de readout macro for state tomography was causing troubles so I fixed as well

### CODE REVIEW

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py [hunk 1]
```diff
@@ -51,52 +52,66 @@
 #####################################
 #  Open Communication with the QOP  #
 #####################################
-qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+# qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+qmm = QuantumMachinesManager()
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:56]: Why did you remove the host?

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py [hunk 2]
```diff
@@ -51,52 +52,66 @@
...
-job = qm.execute(T1)
-res_handles = job.result_handles
-I_handle = res_handles.get("I")
-Q_handle = res_handles.get("Q")
-iteration_handle = res_handles.get("iteration")
-I_handle.wait_for_values(1)
-Q_handle.wait_for_values(1)
-iteration_handle.wait_for_values(1)
-next_percent = 0.1  # First time print 10%
+simulate = True
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:62]: I would put the default to False for all of them (to make them ready to run).

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/allxy.py [hunk 3]
```diff
@@ -303,56 +304,66 @@
 
     with stream_processing():
         for i in range(21):
-            I_st[i].boolean_to_int().average().save(f"I{i}")
-            Q_st[i].boolean_to_int().average().save(f"Q{i}")
+            I_st[i].average().save(f"I{i}")
+            Q_st[i].average().save(f"Q{i}")
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/allxy.py:309]: You sure?

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py [hunk 4]
```diff
@@ -56,7 +56,7 @@ def readout_macro(threshold=None, state=None, I=None, Q=None):
     )
     if threshold is not None:
         assign(state, I > threshold)
-    return state, I, Q
+    return state
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py:60]: Sometimes you need the I & Q, so this needs to stay, we need to fix the scripts.
I actually have a PR which fixes this from about two weeks ago that wasn't merged :\

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/ramsey.py [hunk 5]
```diff
@@ -28,9 +29,9 @@
     with for_(n, 0, n < n_avg, n + 1):
         # Notice it's <= to include t_max (This is only for integers!)
         with for_(tau, tau_min, tau <= tau_max, tau + dtau):
-            play("pi_half", "qubit")
+            # play("x90", "qubit")
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/ramsey.py:32]: Why is this commented?

**Review by yomach — CHANGES_REQUESTED** [review]

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 2 above)
Comment (TheoLaudatQM) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:62]: I think that for teaching purposes it is better to have it to True because this way we can show the sequence before running it

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py [hunk 6]
```diff
@@ -51,52 +52,66 @@
...
+    next_percent = 0.1  # First time print 10%
+
+
+    def on_close(event):
+        event.canvas.stop_event_loop()
+        job.halt()
+
+
+    f = plt.figure()
+    f.canvas.mpl_connect("close_event", on_close)
```
Comment (TheoLaudatQM) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:90]: While this tool is not available in the py qua tools, I would put this function in the configuration file to avoid duplicates.
I am using this function:
```
# Live plotting
def interrupt_on_close(figure, current_job):
    def on_close(event):
        print("Execution stopped by user!")
        current_job.halt()
        event.canvas.stop_event_loop()

    figure.canvas.mpl_connect("close_event", on_close)
```
The function can be called using 
```
fig = plt.figure()
interrupt_on_close(fig, job)  #  Interrupts the job when closing the figure
```

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py [hunk 7]
```diff
@@ -28,7 +29,7 @@
     with for_(n, 0, n < n_avg, n + 1):
         # Notice it's <= to include t_max (This is only for integers!)
         with for_(tau, tau_min, tau <= tau_max, tau + dtau):
-            play("pi", "qubit")
+            play("x180", "qubit")
```
Comment (TheoLaudatQM) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:32]: I must say that this "pi" vs "x180" is a bit confusing to me...
When do we want to use "pi" and when do we want to use "x180"?

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/time_of_flight.py [hunk 8]
```diff
@@ -27,26 +28,36 @@
...
-plt.show()
-
-plt.figure()
-plt.title("Averaged run (Check ToF & DC Offset)")
-plt.plot(adc1)
-plt.plot(adc2)
-plt.show()
-print(f"Input1 mean: {np.mean(adc1)} V\n" f"Input2 mean: {np.mean(adc2)} V")
+# qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+qmm = QuantumMachinesManager()
```
Comment (TheoLaudatQM) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/time_of_flight.py:32]: I believe we still need the IP address here

**Review by TheoLaudatQM — CHANGES_REQUESTED** [review]

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 7 above)
Comment (yomach) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:32]: I would prefer `pi` here, but I didn't bother fighting over it...

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 1 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:56]: This is my bad, to run the program in my laptop I needed to put the empty parenthesis only

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 2 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:62]: I'm on Theo's side

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/allxy.py (hunk 3 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/allxy.py:309]: yes, the measurement statement in this allxy does a simple measurement not state discrimination

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py (hunk 4 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py:60]: ok, well how do we proceed? I  get confused with all of these github methodology now xD

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/ramsey.py (hunk 5 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/ramsey.py:32]: I was testing something and forgot to uncomment

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 6 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:90]: nice

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 7 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:32]: why would you prefer pi? If you use either x180 or y180 you get the same results. Actually in this program indeed you just need to start the state in the excited state, so maybe that tips the code towards 'pi' which is agnostic.

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py (hunk 4 above)
Comment (yomach) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py:60]: I'll help you rebase it on main on Monday

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 7 above)
Comment (yomach) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:32]: Yea, I agree.
There are cases where it is clearly simpler to use `x180` and notation, for example, randomized benchmarking.
On the contrary, if you use frame rotation, it is clearer to use `pi`.
The question is what happens in the "simple" experiments such as Ramsey or T1.
I think that the fact that you don't care about the axis makes it better to use `pi`. This is what you won't say if you would describe the experiment in words.

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py (hunk 4 above)
Comment (yomach) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py:60]: Try to address the rest of the comments here and next week we'll do the github stuff together.

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/time_of_flight.py (hunk 8 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/time_of_flight.py:32]: yes, this was my mistake

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py (hunk 7 above)
Comment (KevinAVR) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/T1.py:32]: I like 'pi' in T1 experiments and when using frame rotations that is what I am going to do going forward.

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py [hunk 9]
```diff
@@ -97,10 +97,12 @@ def readout_macro(threshold=None, state=None, I=None, Q=None):
 x90_wf, x90_der_wf = np.array(drag_gaussian_pulse_waveforms(x90_amp, x90_len, x90_sigma, alpha=0, delta=1))
 # No DRAG when alpha=0, it's just a gaussian.
 
-x270_len = x180_len
-x270_sigma = x270_len / 5
-x270_amp = -x90_amp
-x270_wf, x270_der_wf = np.array(drag_gaussian_pulse_waveforms(x270_amp, x270_len, x270_sigma, alpha=0, delta=1))
+minus_x90_len = x180_len
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/configuration.py:100]: I thought that there was an agreement to call it `x270` and not `minus_x90`.
I don't really care what we choose, but:
1. Let's make a decision because you'll change it to `minus_x90` and tomorrow someone would want to change it back.
2. We need it to be consistent amount all of our QCAs

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/echo.py [hunk 10]
```diff
@@ -31,11 +32,11 @@
     with for_(n, 0, n < n_avg, n + 1):
         # Notice it's <= to include t_max (This is only for integers!)
         with for_(tau, tau_min, tau <= tau_max, tau + dtau):
-            play("pi_half", "qubit")
+            play("x90", "qubit")
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/echo.py:35]: didn't we say to keep it pi here?

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/qubit_spec.py [hunk 11]
```diff
@@ -56,58 +58,71 @@
...
 
-    plt.plot(freqs, np.sqrt(I**2 + Q**2), ".")
-    # plt.plot(freqs + qubit_LO, np.sqrt(I**2 + Q**2), '.')
+    while res_handles.is_processing():
+        plt.cla()
+        I = I_handle.fetch_all()
+        Q = Q_handle.fetch_all()
+        iteration = iteration_handle.fetch_all()
+        # If we want to plot the phase
+        phase = signal.detrend(np.unwrap(np.angle(I + 1j * Q)))
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/qubit_spec.py:100]: qubit spect don't need the `signal.detrend` because the readout frequency is constant (so there is no linear trend)

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/IQ_blobs.py [hunk 12]
```diff
@@ -39,8 +38,9 @@
         save(Q_g, Q_g_st)
         wait(cooldown_time, "resonator")
 
-        align()
-        play("pi", "qubit")
+        align()  # global align
+
+        play("x180", "qubit")
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/IQ_blobs.py:43]: I think we said to keep it as pi here, no?

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/state_tomography.py [hunk 13]
```diff
@@ -26,16 +25,15 @@
     with for_(n, 0, n < n_avg, n + 1):
         with for_(c, 0, c <= 2, c + 1):
             # Add here whatever state you want to characterize
-
             with switch_(c):
                 with case_(0):  # projection along X
-                    play("-Y/2", "qubit")
+                    play("y270", "qubit")
```
Comment (yomach) [inline Quantum-Control-Applications/Superconducting/Single Fixed Transmon/state_tomography.py:30]: here you wrote `y270`... should be `-x90` (unless we change it back)

File: Quantum-Control-Applications/Superconducting/Single Fixed Transmon/allxy.py [hunk 14]
```diff
@@ -307,56 +308,65 @@
...
+        ("x90", "y90"),
+        ("y90", "x90"),
+        ("x90", "y180"),
+        ("y90", "x180"),
+        ("x180", "y90"),
+        ("y180", "x90"),
+        ("x90", "x180"),
+        ("x180", "x90"),
+        ("y90", "y180"),
+        ("y", "y90"),
```
[the remark this answers is not in the document]
Comment (TheoLaudatQM) [reply Quantum-Control-Applications/Superconducting/Single Fixed Transmon/allxy.py:365]: I think it should be ` ("y180", "y90")`

**Comment by yomach** [comment]
@TheoLaudatQM can you please approve?

**Review by TheoLaudatQM — APPROVED** [review]
